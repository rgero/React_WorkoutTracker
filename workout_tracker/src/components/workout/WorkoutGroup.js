import React from 'react';
import Accordion from 'react-bootstrap/Accordion';
import { WorkoutListItem } from "./WorkoutListItem";
import { GetMonthName, GetOrderedValues } from '../../helpers/WorkoutListProcesser';

const WorkoutGroup = ({workoutList, descendingOrder}) => {
    let months = GetOrderedValues(workoutList, descendingOrder);
    return (
        <div>
            {
                months.map((month, index)=> (
                    <Accordion defaultActiveKey={index} key={`WorkoutGroup_${index}`}>   
                        <Accordion.Item>
                            <Accordion.Header>{GetMonthName(month)}</Accordion.Header>
                            <Accordion.Body>
                                {
                                    workoutList[month].map((workout, index) => (
                                            <WorkoutListItem workout={workout} key={`WorkoutListItem_${index}`} />
                                        ))
                                }
                            </Accordion.Body>
                        </Accordion.Item>
                    </Accordion>
                ))
            }
        </div>
      );
}

export default WorkoutGroup;