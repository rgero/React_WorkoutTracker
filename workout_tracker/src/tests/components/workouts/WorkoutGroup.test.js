import TestRenderer from 'react-test-renderer';

import WorkoutGroup from '../../../components/workout/WorkoutGroup';
import { TestWorkouts } from '../../testData';

jest.mock('../../../components/workout/WorkoutListItem', () => ({
    WorkoutListItem: (workout, index)=> {
        return (<div>{workout.workout.workoutDate}</div>);
    }
}));

describe("Render tests", ()=> {

    let testWorkouts = {
        "02": [TestWorkouts[1], TestWorkouts[3]]
    }

    test("Data descending, matches snapshot", ()=> {
        let element = new TestRenderer.create(
            <WorkoutGroup workoutList={testWorkouts} descending={true}/>
        )
        expect(element.toJSON()).toMatchSnapshot();
    })

    test("Data ascending, matches snapshot", ()=> {

        let element = new TestRenderer.create(
            <WorkoutGroup workoutList={testWorkouts} descending={false}/>
        )
        expect(element.toJSON()).toMatchSnapshot();
    })

})