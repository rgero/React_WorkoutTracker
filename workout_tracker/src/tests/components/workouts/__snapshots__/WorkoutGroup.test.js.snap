// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render tests Data ascending, matches snapshot 1`] = `
<div
  className="accordion"
>
  <div
    className="accordion-item"
  >
    <h2
      className="accordion-header"
    >
      <button
        aria-expanded={false}
        className="accordion-button collapsed"
        onClick={[Function]}
        type="button"
      >
        February
      </button>
    </h2>
    <div
      aria-expanded={null}
      className="accordion-collapse collapse"
    >
      <div
        className="accordion-body"
      >
        <div>
          2022-02-08T00:00:00.000Z
        </div>
        <div>
          2022-03-15T00:00:00.000Z
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render tests Data descending, matches snapshot 1`] = `
<div
  className="accordion"
>
  <div
    className="accordion-item"
  >
    <h2
      className="accordion-header"
    >
      <button
        aria-expanded={false}
        className="accordion-button collapsed"
        onClick={[Function]}
        type="button"
      >
        February
      </button>
    </h2>
    <div
      aria-expanded={null}
      className="accordion-collapse collapse"
    >
      <div
        className="accordion-body"
      >
        <div>
          2022-02-08T00:00:00.000Z
        </div>
        <div>
          2022-03-15T00:00:00.000Z
        </div>
      </div>
    </div>
  </div>
</div>
`;
